cdc/generation.cc

```diff
@@ -59,14 +59,57 @@ static void copy_int_to_bytes(int64_t i, size_t offset, bytes& b) {
     std::copy_n(reinterpret_cast<int8_t*>(&i), sizeof(int64_t), b.begin() + offset);
 }
 
-stream_id::stream_id(int64_t first, int64_t second)
+static constexpr auto stream_id_version_bits = 4;
+static constexpr auto stream_id_random_bits = 38;
```

Comment:
I'm making some nitpicks here, but you can consider them for a followup patch - Seeing that @haaawk and @kbr- have already oked this patch I'm planning to commit it unless I find a big hole in it.
My first nitpick is that you should perhaps have a comment that the random bits vs index bits is true for version 0 (or whatever you call it) and may change in a future version.
My second nitpick is that I have no idea what "index" refers to, this probably deserves a comment.

Reply:
My approach to bits changing is changing the constant names to `stream_id_version_bits_v1` and `stream_id_version_bits_v2` _when_ the change happens, but keep it as a singular constant even by name until then, to not mislead someone to think there is another one somewhere until there is. 

Index == vnode index, commented elsewhere, but could maybe use a few extra.